mailcow_integration/squire_mailcow.py

```diff
@@ -0,0 +1,338 @@
+        It is responsible for adding members' emails to a group of member aliases (depending
+        on their stored preferences), and adding members' emails to the aliases of committees
+        they are in.
+
+        When this class interacts with the Mailcow API, it leaves behind some form of
+        "MANAGED BY SQUIRE" comment. If this class sees an already existing item in the API it
+        wishes to modify, it will only do so if this comment is present. This prevents it from
+        accidentally overwriting information that a Mailcow admin may have added. Furthermore,
+        it also allows manual overrides by Mailcow admins.
+    """
+    SQUIRE_MANAGE_INDICATOR = "[MANAGED BY SQUIRE]"
+    INTERNAL_ALIAS_SETTING_NAME = "%s Internal Alias" % SQUIRE_MANAGE_INDICATOR
+    ALIAS_COMMITTEE_PUBLIC_COMMENT = "%s Committee Alias" % SQUIRE_MANAGE_INDICATOR
+    ALIAS_MEMBERS_PUBLIC_COMMENT = "%s Members Alias" % SQUIRE_MANAGE_INDICATOR
+    ALIAS_GLOBAL_COMMITTEE_PUBLIC_COMMENT = "%s Global Committee Alias" % SQUIRE_MANAGE_INDICATOR
+
+    def __init__(self, mailcow_host: str, mailcow_api_key: str):
+        self._client = MailcowAPIClient(mailcow_host, mailcow_api_key)
+
+        # Cannot import directly because this class is initialized before the app registry is set up
+        self._member_model = apps.get_model("membership_file", "Member")
+        self._committee_model = apps.get_model("committees", "AssociationGroup")
+        self._user_preferences_model = apps.get_model("dynamic_preferences_users", "UserPreferenceModel")
+
+        # List of internal addresses (sorted in order of appearance in the config)
+        self.INTERNAL_ALIAS_ADDRESSES: List[str] = [
+            address for address, alias_data in settings.MEMBER_ALIASES.items() if alias_data['internal']
+        ] + settings.COMMITTEE_CONFIGS["global_addresses"]
+
+        # List of all member alias addresses; these should never be included in an alias's goto addresses
+        self.BLOCKLISTED_EMAIL_ADDRESSES: List[str] = [
+            address for address in settings.MEMBER_ALIASES.keys()
+        ] + settings.COMMITTEE_CONFIGS["global_addresses"]
+
+        # Caches
+        self._alias_cache: Optional[List[MailcowAlias]] = None
+        self._mailbox_cache: Optional[List[MailcowMailbox]] = None
+        self._alias_map_cache: Optional[Dict[str, MailcowAlias]] = None
+        self._mailbox_map_cache: Optional[Dict[str, MailcowMailbox]] = None
+
+    @property
+    def mailcow_host(self):
+        return self._client.host
+
+    def __str__(self) -> str:
+        return f"SquireMailcowManager[{self.mailcow_host}]"
+
+    def clean_emails(self, queryset: QuerySet, email_field="email", extra: Optional[List[str]]=None) -> QuerySet:
+        """ Cleans a queryset of models with an email field (of any name),
+            by removing blocklisted email addresses.
+            Additional blocklisted addresses can be passed through the
+            `extra` keyword argument.
```

Comment:
Extra is removed from the queryset cleaned list, which is counterintuitive to the word extra. Maybe use the word exclude instead?